### PR TITLE
App updater: download to usb0 if available

### PR DESCRIPTION
**Pull request by illusionyy**

### CODE REVIEW

**Comment by bucanero** [comment]
question, any specific reason or use-case for downloading the update to USB? 🤔

**Comment by illusionyy** [comment]
Normal pkg installer from debug settings required usb device. Maybe it's better to implement daemon to install the update instead

**Comment by bucanero** [comment]
> Normal pkg installer from debug settings required usb device. Maybe it's better to implement daemon to install the update instead

right, I was assuming that all users would be using latest goldhen right now (so `/data/pkg` shows up) but yes if they're using something old, then only .pkg on USB will show to install.

File: source/settings.c [hunk 1]
```diff
@@ -200,11 +200,20 @@ void update_callback(int sel)
...
-		char* download_path = (dir_exists("/data/pkg/") == SUCCESS) ? "/data/pkg/goldcheats.pkg" : "/data/goldcheats.pkg";
+		char* download_path = "/data/goldcheats.pkg";
+		if (dir_exists("/mnt/usb0/") == SUCCESS)
+		{
+			download_path = "/mnt/usb0/goldcheats.pkg";
+		}
+		else if (dir_exists("/data/pkg/") == SUCCESS)
+		{
+			download_path = "/data/pkg/goldcheats.pkg";
+		}
```
Comment (bucanero) [inline source/settings.c]:
```suggestion
		char* download_path = (dir_exists("/data/pkg/") == SUCCESS) ? "/data/pkg/goldcheats.pkg" : "/data/goldcheats.pkg";
		if (dir_exists("/mnt/usb0/") == SUCCESS)
		{
			download_path = "/mnt/usb0/goldcheats.pkg";
		}
```

**Review by bucanero — APPROVED** [review]

**Pull request merged**